### PR TITLE
Switching notifications to FCM over (now deprecated) GCM

### PR DESCRIPTION
**Pull request by tweirtx**
## Description

Changed the Google Cloud Messaging send URL used for notifications to the Firebase Cloud Messaging one, since GCM is now deprecated and will go offline during next season.
## Motivation and Context

GCM is being deprecated, so this is migrating to its successor. If this change is not made then notifications will stop working. Solves issue #2268
I have also changed the authkeys page to reflect that it is now the FCM key, not the GCM key.
## How Has This Been Tested?

Phil tested this with a test version of the Android app and was able to send a test notification to himself with it.
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change API specifications or require data migrations)

### CODE REVIEW

**Comment by bdaroz** [comment]
Do we have any testing on sending FCM messages to GCM clients, or will this break notifications on GCM clients? (Eg existing, not updated, Android app)

**Comment by bdaroz** [comment]
Disregard my last - messages crossed in the ether - I just saw your comment on #2268 .
👍

**Comment by phil-lopreiato** [comment]
This should hopefully be safe, I deployed it to my test setup and made sure I could still send push notifications to my dev app build.

That being said, I'd still like some help testing this out in prod after it gets merged :)

**Comment by bdaroz** [comment]
Do we have a way to send notifications in prod w/o an active event?

**Comment by phil-lopreiato** [comment]
@bdaroz yes, the ping feature as well as the webhook test page have this ability

File: controllers/gcm/gcm.py [hunk 1]
```diff
@@ -104,12 +104,11 @@ def __init__(self):
...
         self.SERVER_KEY = Sitevar.get_by_id('gcm.serverKey')
         if self.SERVER_KEY is None:
-            raise Exception("Missing sitevar: gcm.serverKey. Can't send GCM messages.")
+            raise Exception("Missing sitevar: gcm.serverKey. Can't send FCM messages.")
         self.GCM_CONFIG = {'gcm_api_key': self.SERVER_KEY.contents['gcm_key']}
         self.GOOGLE_LOGIN_URL = 'https://www.google.com/accounts/ClientLogin'
         # Can't use https on localhost due to Google cert bug
-        self.GOOGLE_GCM_SEND_URL = 'https://android.apis.google.com/gcm/send'
-        self.GOOGLE_GCM_SEND_URL = 'https://android.googleapis.com/gcm/send'
+        self.GOOGLE_GCM_SEND_URL = 'https://fcm.googleapis.com/fcm'
```
Comment (phil-lopreiato) [inline controllers/gcm/gcm.py]: Whoops, this should be `https://fcm.googleapis.com/fcm/send`

Fixed in 3c72a6d1d90510192b3bfa68d8678691f776ebb7

**Pull request merged**